### ISSUE TITLE
Talos v1.3 with Cilium enabled and Kube-Proxy Disabled Causes CoreDNS startup failure

### ISSUE THREAD

**Issue by claughinghouse** [siderolabs/talos#6939]
### Description

Following the process to deploy talos without kube-proxy and Cilium per the [docs](https://www.talos.dev/v1.3/kubernetes-guides/network/deploying-cilium/#method-4-helm-manifests-inline-install) it is advised to set `cluster.proxy.disabled: true`, and then appending the Cilium to the inline config key. 

When deployed, CoreDNS fails to become healthy with complaints that kube-api is signed by an untrusted CA. 

[Deployed talosctl configuration](https://gist.github.com/claughinghouse/f237ccc43a2d4e7567868195a9447d6e)

### Logs

```
[WARNING] plugin/kubernetes: Kubernetes API connection failure: Get "[https://10.96.0.1:443/version](https://10.96.0.1/version)": x509: certificate signed by unknown authority

[INFO] plugin/ready: Still waiting on: "kubernetes"
```

### Environment

- Talos version: 
```
~  talosctl version --nodes 10.1.101.200
Client:
	Tag:         v1.3.5
	SHA:         03edf8c1
	Built:
	Go version:  go1.19.6
	OS/Arch:     darwin/amd64
Server:
	NODE:        10.1.101.200
	Tag:         v1.3.5
	SHA:         03edf8c1
	Built:
	Go version:  go1.19.6
	OS/Arch:     linux/amd64
	Enabled:     RBAC
```
- Kubernetes version: 
```
kubectl version --short
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.26.2
Kustomize Version: v4.5.7
Server Version: v1.26.1
```
- Platform: Talos v1.3 on ProxmoxVM.

**Comment by smira** [comment]
Have you checked updated docs? https://www.talos.dev/v1.4/kubernetes-guides/network/deploying-cilium/

**Comment by claughinghouse** [comment]
I did not since v1.13 is stable. I'll go through the process this week. 

[Link to conversation with @frezbo](https://taloscommunity.slack.com/archives/CMARMBC4E/p1678562211182049?thread_ts=1678549050.728489&channel=CMARMBC4E&message_ts=1678562211.182049)

**Comment by jonkerj** [comment]
I believe you are running into cilium/cilium#21603

**Comment by claughinghouse** [comment]
> I believe you are running into [cilium/cilium#21603](https://github.com/cilium/cilium/issues/21603)

Yep. After implementing [this comment on that issue](https://github.com/cilium/cilium/issues/21603#issuecomment-1453753820) I had success.
